out_writeapi.go

```diff
@@ -174,48 +227,20 @@ func FLBPluginInit(plugin unsafe.Pointer) int {
 	tableID := output.FLBPluginConfigKey(plugin, "TableID")
 
 	//optional maxchunksize param
-	str := output.FLBPluginConfigKey(plugin, "Max_Chunk_Size")
-	var maxChunkSize_init int
-	if str != "" {
-		maxChunkSize_init, err = strconv.Atoi(str)
-		if err != nil {
-			log.Printf("Invalid Max Chunk Size, defaulting to 9 MB:%v", err)
-			maxChunkSize_init = 9 * 1024 * 1024
-		}
-		if maxChunkSize_init > 9*1024*1024 {
-			log.Println("A single call to AppendRows cannot exceed 9 MB.")
-			maxChunkSize_init = 9 * 1024 * 1024
-		}
+	maxChunkSize_init := getConfigField(plugin, "Max_Chunk_Size", "Invalid Max Chunk Size, defaulting to 9 MB", (9 * 1024 * 1024))
+	if maxChunkSize_init > 9*1024*1024 {
+		log.Println("A single call to AppendRows cannot exceed 9 MB.")
+		maxChunkSize_init = 9 * 1024 * 1024
 	}
 
 	//optional max queue size params
-	str1 := output.FLBPluginConfigKey(plugin, "Max_Queue_Requests")
-	str2 := output.FLBPluginConfigKey(plugin, "Max_Queue_Bytes")
-	var queueSize int
-	var queueByteSize int
-	if str1 == "" {
-		queueSize = 1000
-	} else {
-		queueSize, err = strconv.Atoi(str1)
-		if err != nil {
-			log.Printf("Invalid Max Queue Requests, defaulting to 1000:%v", err)
-			queueSize = 1000
-		}
-	}
-	if str2 == "" {
-		queueByteSize = (100 * 1024 * 1024)
-	} else {
-		queueByteSize, err = strconv.Atoi(str2)
-		if err != nil {
-			log.Printf("Invalid Max Queue MB, defaulting to 100 MB:%v", err)
-			queueByteSize = (100 * 1024 * 1024)
-		}
-	}
+	queueSize := getConfigField(plugin, "Max_Queue_Requests", "Invalid Max Queue Requests, defaulting to 1000", 1000)
+	queueByteSize := getConfigField(plugin, "Max_Queue_Bytes", "Invalid Max Queue Bytes, defaulting to 100 MB", (100 * 1024 * 1024))
 
 	//create new client
-	client, err := managedwriter.NewClient(ms_ctx, projectID)
+	client, err := getClient(ms_ctx, projectID)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("getClient failed in FLBPlugininit: %s", err)
```

Comment:
This line won't terminate the program, right?

Reply:
It will. If we can't get the client wouldn't we want to terminate the program? I can also work on better error handling (switching to log.Printf and returning FLB_ERROR instead of log.Fatalf) in another commit.